### PR TITLE
Add config level for log4jlogger [GEOT-5625]

### PR DESCRIPTION
**Pull request by jodygarnett**
This will allow config to be used, it ends up one step below INFO level.

* https://osgeo-org.atlassian.net/browse/GEOT-5625

### CODE REVIEW

File: modules/library/metadata/src/main/java/org/geotools/util/logging/Log4JLogger.java [hunk 1]
```diff
@@ -33,6 +32,10 @@
  * @see Logging
  */
 final class Log4JLogger extends LoggerAdapter {
+    
+    private static final int CONFIG_INT = org.apache.log4j.Level.INFO_INT-1;
+    private static final org.apache.log4j.Level CONFIG = org.apache.log4j.Level.toLevel( CONFIG_INT, org.apache.log4j.Level.TRACE );
```
Comment (jodygarnett) [inline modules/library/metadata/src/main/java/org/geotools/util/logging/Log4JLogger.java]: default level should be DEBUG here (sigh!)

Comment (aaime) [inline modules/library/metadata/src/main/java/org/geotools/util/logging/Log4JLogger.java]: Needs some test and docs for config log4j

**Comment by jodygarnett** [comment]
Putting this on hold until after code freeze.

**Comment by bencaradocdavies** [comment]
Code freeze is over.

**Comment by jodygarnett** [comment]
Does anyone care to persue this? Would love to have less verbose startup.

**Comment by jodygarnett** [comment]
Attaching to bug ticket and closing

**Pull request closed without merging**